core/templates/dev/head/domain/exploration/QuestionListObjectFactorySpec.js

```diff
@@ -158,6 +158,21 @@ describe('Question List Object Factory', function() {
     ).toBe(true);
   });
 
+  it('should remove the given question if it exists', function() {
+    var questionToDelete = sampleQuestions[1];
+    questionList.removeQuestion(questionToDelete);
+    expect((function() {
```

Comment:
This is way too much logic for a test. Just do explicit equality asserts for the length of questionList.getQuestions() and the state names for each of the remaining questions.